packages/reducers/src/makeEnhancer.test.js

```diff
@@ -44,7 +46,18 @@ describe('makeEnhancer', () => {
 		expect(getEntries(store)).toEqual([]);
 	});
 
-	it('throws when injecting a function without a namespace', () => {
-		expect(() => store.injectReducers(identity)).toThrow();
+	it('removes data from state after reducer ejecting', () => {
+		const store = createStoreRedux(identity, makeEnhancer());
+		const reducerA = (state = { name: 'a' }) => state;
+
+		store.injectReducers({ a: reducerA }, { namespace: 'nsA' });
+		expect(store.getState()).toEqual({
+			namespaces: {
+				nsA: { a: { name: 'a' } },
+			},
+		});
+
+		store.ejectReducers(identity, { namespace: 'nsB' });
```

Comment:
nooo, this should not be the expected behaviour.

the state clean-up should only happen for the appropriate reducer

	store.ejectReducers(identity, { namespace: 'nsB' });

this code should not do anything to our state, because identity was never present in the store!

what we need to do is check that

```store.ejectReducers({ a: reducerA }, { namespace: 'nsA' });```

doesnt leave any traces of the ejected reducer's state

Reply:
change this test and please add another test to ensure that the state of other reducers is left untouched